src/Tests/Tests/ClientConcepts/ConnectionPooling/BuildingBlocks/ConnectionPooling.doc.cs

```diff
@@ -79,6 +82,88 @@ public class ConnectionPooling
 			client.ConnectionSettings.ConnectionPool
 				.Should().BeOfType<SingleNodeConnectionPool>();
 		}
+		
+		/**
+		* [[cloud-connection-pool]]
+		* ==== CloudConnectionPool
+		*
+		* A specialized subclass of `SingleNodeConnectionPool that accepts a Cloud Id and credentials.
+		* When used the client will also pick Elastic Cloud optmized defaults for the connection settings.
+		 *
+		 * A Cloud Id for your cluster can be fetched from your Elastic Cloud cluster administration console.
+		 *
+		 * A Cloud Id should be in the form of `cluster_name:base_64_data` where `base_64_data` are the UUIDs for the services in this cloud instance e.g
+		 *
+		 * `host_name$elasticsearch_uuid$kibana_uuid$apm_uuid`
+		 *
+		 * Out of these only `host_name` and `elasticsearch_uuid` are always available.
+		 * 
+		*/
+		[U] public void CloudConnectionPool()
+		{
+			string ToBase64(string s) => Convert.ToBase64String(Encoding.UTF8.GetBytes(s));
+			/* Here we obviously use a ficticuous Cloud Id so lets create a fake one. */
+
+			var hostName = "cloud-endpoint.example";
+			var elasticsearchUuid = "3dadf823f05388497ea684236d918a1a";
+			var services = $"{hostName}${elasticsearchUuid}$3f26e1609cf54a0f80137a80de560da4";
+			var cloudId = $"my_cluster:{ToBase64(services)}";
+
+			/*
+			 * In a real scenario you would be able to copy paste `cloudId`
+			 *
+			 * A cloud connection pool always needs credentials as well here opt for basic auth 
+			 */
+			var credentials = new BasicAuthenticationCredentials("username", "password");
+			var pool = new CloudConnectionPool(cloudId, credentials);
+			pool.UsingSsl.Should().BeTrue();
```

Comment:
I think we should start steering away from emitting test assertions in _documentation_. I think it adds noise to the signal we're trying to display to users, in this case, how to configure the client for user with Elasticsearch Service on Elastic Cloud. We can prefix and assertions with `// hide`, and introduce a block scope `{}` and prefix, if needing to hide more.